src/crimson/os/seastore/segment_manager.cc

```diff
@@ -63,18 +63,26 @@ SegmentManager::get_segment_manager(
 LOG_PREFIX(SegmentManager::get_segment_manager);
   return seastar::do_with(
     static_cast<size_t>(0),
-    [&](auto &nr_zones) {
+    [FNAME,
+     dtype,
+     &device](auto &nr_zones) {
       return seastar::open_file_dma(
 	device + "/block",
 	seastar::open_flags::rw
-      ).then([&](auto file) {
+      ).then([FNAME,
+	      dtype,
+	      &device,
+	      &nr_zones](auto file) {
 	return seastar::do_with(
 	  file,
-	  [=, &nr_zones](auto &f) -> seastar::future<int> {
+	  [&nr_zones](auto &f) -> seastar::future<int> {
 	    ceph_assert(f);
 	    return f.ioctl(BLKGETNRZONES, (void *)&nr_zones);
 	  });
-      }).then([&](auto ret) -> crimson::os::seastore::SegmentManagerRef {
+      }).then([FNAME,
+	       dtype,
+	       &device,
```

Comment:
Only `nr_zones` is extended by `do_with()`, the `device` string needs to be copied.

Reply:
Ok, updated the patch.